flight/core/Dispatcher.php

```diff
@@ -105,42 +123,59 @@ public function has(string $name): bool
      */
     public function clear(?string $name = null): void
     {
-        if (null !== $name) {
+        if ($name !== null) {
             unset($this->events[$name]);
             unset($this->filters[$name]);
-        } else {
-            $this->events = [];
-            $this->filters = [];
+
+            return;
         }
+
+        $this->events = [];
+        $this->filters = [];
     }
 
     /**
      * Hooks a callback to an event.
      *
-     * @param string   $name     Event name
-     * @param string   $type     Filter type
-     * @param callable $callback Callback function
+     * @param string $name Event name
+     * @param 'before'|'after' $type Filter type
+     * @param Closure(array<int, mixed> &$params, string &$output): (void|false) $callback
+     *
+     * @return $this
      */
-    public function hook(string $name, string $type, callable $callback): void
+    public function hook(string $name, string $type, callable $callback): self
     {
+        if (!in_array($type, self::FILTER_TYPES, true)) {
```

Comment:
If you want to take this one step further `in_array($type, self::FILTER_TYPES, true) === false` is actually slightly more performent because `!` is a falsey evaluation and it has to evaluate multiple potential false values (false, 0, '', etc)

Reply:
I'll fix it :)I'll fix it :)

Reply:
[`in_array(mixed $needle, array $haystack, bool $strict = false): bool`]()https://www.php.net/manual/es/function.in-array.php).

That's because I can use ! safety that in_array always returns a bool.